spinalcordtoolbox/straightening.py

```diff
@@ -518,31 +520,26 @@ def straighten(self):
         image_centerline_straight.save(fname_ref)
         if self.curved2straight:
             logger.info('Apply transformation to input image...')
-            run_proc(['isct_antsApplyTransforms',
-                      '-d', '3',
-                      '-r', fname_ref,
-                      '-i', 'data.nii',
-                      '-o', 'tmp.anat_rigid_warp.nii.gz',
-                      '-t', 'tmp.curve2straight.nii.gz',
-                      '-n', 'BSpline[3]'],
-                     is_sct_binary=True,
-                     verbose=verbose)
+            sct_apply_transfo.main(['-i', 'data.nii',
+                                    '-d', fname_ref,
+                                    '-w', 'tmp.curve2straight.nii.gz',
+                                    '-o', 'tmp.anat_rigid_warp.nii.gz',
+                                    '-x', 'spline',
+                                    '-v', str(verbose)])
+
```

Comment:
I [reverted the change above](https://github.com/neuropoly/spinalcordtoolbox/commit/b7909506fa6a9f5a72e1227fcd4ec16cf79b7972), and the `batch_processing.sh` [tests pass](https://github.com/neuropoly/spinalcordtoolbox/runs/2088187862). So, I believe `straightening.py` is the cause.

### Investigation

In `batch_processing.sh`, the highlighted call is used in two scripts:

https://github.com/neuropoly/spinalcordtoolbox/blob/0b7be07b7e64a57a303824e85930fbceb8b890f0/batch_processing.sh#L73

https://github.com/neuropoly/spinalcordtoolbox/blob/0b7be07b7e64a57a303824e85930fbceb8b890f0/batch_processing.sh#L79

In both calls, `straight_ref.nii.gz` (which is just a renamed `tmp.anat_rigid_warp.nii.gz`) is used in many subsequent steps (its header gets copied as a reference for other output images). So, I ran the script twice (before/after changes) to compare the headers for each straightened file: https://www.diffchecker.com/GbBSFdZ0

The headers are the only thing that differ (the data was the same). So, my conclusion is that the shift in `batch_processing.sh` values is not a bad thing, because  `straight_ref.nii.gz` has had the wrong headers this whole time. (Same issue as #3254.)

Reply:
> So, I ran the script twice (before/after changes) to compare the headers for each straightened file: https://www.diffchecker.com/GbBSFdZ0

The only strange thing here is the sform/qform codes for both straightened files. Neither looks quite right to me.

Reply:
There are suspicious lines under `image.py`, e.g.

https://github.com/neuropoly/spinalcordtoolbox/blob/60ea08c9d664cf403f2982f8f5c8ae1cc5d704a3/spinalcordtoolbox/image.py#L1417

Relevant doc: https://nipy.org/nibabel/nifti_images.html#default-sform-and-qform-codes
![image](https://user-images.githubusercontent.com/2482071/111332803-7abe6f00-8648-11eb-8b89-ea1bd2dfa0dd.png)

Analogous problem as in https://github.com/ivadomed/ivadomed/pull/714

Moreover, I see this line: 
https://github.com/neuropoly/spinalcordtoolbox/blob/60ea08c9d664cf403f2982f8f5c8ae1cc5d704a3/spinalcordtoolbox/image.py#L1330

Which, according to the doc, might be problematic if qform/sform are different:
![image](https://user-images.githubusercontent.com/2482071/111332997-a3466900-8648-11eb-9089-994892c861eb.png)

Reply:
Ahh! Thank you for sharing what was learned in https://github.com/ivadomed/ivadomed/pull/714. You've saved me from having to figure this out all over again. :smile:

Reply:
Perhaps a separate SCT issue should be created for this? It seems this issue with `image.py` has larger implications outside the scope of this PR (which is addressing uninitialized sforms from `isct_antsApplyTransform`).

Reply:
Issue here: https://github.com/neuropoly/spinalcordtoolbox/issues/3295